flask/app.py

```diff
@@ -1029,6 +1029,27 @@ def index():
         rule = self.url_rule_class(rule, methods=methods, **options)
         rule.provide_automatic_options = provide_automatic_options
 
+        for prior_rule in self.url_map.iter_rules():
```

Comment:
I would love to not have to iterate over all rules but `WerkZeug.Map` only has `iter_rules` as a search mechanism. Another potential is that we add the tuple that we care about `(subdomain, method, url_path)` to a set and see if there's existence in the set. I wasn't sure if it would be better to add a new class attribute or to only use existing structures.